### PR TITLE
[ci] Add test to ensure mocks are up-to-date

### PR DESCRIPTION
**Pull request by Dimitar-St**
Linked with this issue #334. Added workflow for keeping mocks up to date.

### CODE REVIEW

File: .github/workflows/auto-generated-checker.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '1.20'
+          check-latest: true
+      - uses: bufbuild/buf-setup-action@v1.23.1
+      - shell: bash
+        run: scripts/protobuf_codegen.sh
```
Comment (patrick-ogrady) [inline .github/workflows/auto-generated-checker.yml:23]: We should remove this, there is no protobuf in this repo.

File: .github/workflows/auto-generated-checker.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Auto-Generated Code Checker
+on:
```
Comment (patrick-ogrady) [inline .github/workflows/auto-generated-checker.yml:2]: Last change...can you make this `on:` section match: https://github.com/ava-labs/hypersdk/blob/580b03c0b4f9b00c95bbd00a699e2c019e09d128/.github/workflows/hypersdk-static-analysis.yml#L6-L10

Comment (Dimitar-St) [reply .github/workflows/auto-generated-checker.yml:2]: Fixed. Really foolish mistakes, I apologize!

Comment (patrick-ogrady) [reply .github/workflows/auto-generated-checker.yml:2]: No worries! Running now.

**Comment by patrick-ogrady** [comment]
Going to fix after merging

**Comment by Dimitar-St** [comment]
> Going to fix after merging

Is there a specific reason why it's failing with permission denied ?

**Comment by patrick-ogrady** [comment]
> > Going to fix after merging
> 
> Is there a specific reason why it's failing with permission denied ?

Yeah, you didn't `chmod` the script.

**Pull request merged**